watcher/watcher.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"context"
+	"github.com/bcc-code/mediabank-bridge/log"
+	"github.com/samber/lo"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+)
+
+type watcher struct {
+	path            string
+	interval        time.Duration
+	recentlyUpdated []string
+	lastUpdated     time.Time
+	callbackUrl     string
+}
+
+func newWatcher(path string, interval time.Duration, callbackUrl string) *watcher {
+	return &watcher{
+		interval:    interval,
+		path:        path,
+		lastUpdated: time.Now(),
+		callbackUrl: callbackUrl,
+	}
+}
+
+func listDirectory(path string) []string {
+	var result []string
+	entries, _ := os.ReadDir(path)
```

Comment:
Would it make sense to replace this with https://pkg.go.dev/path/filepath#Glob ?
Then you could pass in a pattern and only handle for example `*.mov` files easily.